docs/cip/README.md

```diff
@@ -461,247 +1988,671 @@ The Leios protocol may have to mitigate the following categories of threats.
-| 2030 |              ~50% |              24,000,000 |                  41.9 |          0.60 |
+![SPO profitability under Leios.](images/leios-forecast-sqrt-fill.svg)
 
-Note that by 2029, to compensate for Reserve depletion, the network would need to process approximately 36 TPS with average-sized transactions, requiring an Input Block rate of around 0.52 IB/s, roughly 10 times the current mainnet throughput. Leios's design would comfortably support this increased throughput while maintaining decentralization.
+<em>Figure 17: SPO profitability under Leios</em>
+
+<a id="intensive-plutus-execution" href="#intensive-plutus-execution">**Intensive
+Plutus execution**</a>
+
+Because there typically is a time window of several seconds from the time a
+Leios endorser block can be created to when voting must complete, there is also
+an opportunity to do more computation validating transactions within an endorser
+block than for a Praos ranking block. This opens the possibility of increasing
+the Plutus execution budget for endorser blocks so that it is significantly
+larger than the budget for Praos blocks. At the very least a script could be
+allowed to use the whole Plutus execution budget for an endorser block, instead
+of just one quarter of it as is the case for Praos.
+
+Numerous emerging use cases on Cardano would benefit from larger Plutus
+execution budgets. Complex dapps currently have to split a single logical
+operation into a sequence of several transactions, increasing the development
+effort, the complexity, and the attack surface of the scripts involved.
+
+- _ZK proofs:_ It may be possible to increase the Plutus execution budget enough
+  that a complete ZK proof could verified in a single transaction.
+- _Large number of parties:_ Scripts managing potential interactions with a
+  large number of parties (e.g., airdrops, lotteries, and local accounts) are
+  intrinsically limited by Plutus execution limits.
+- _On-chain interpreters:_ Dapps like Marlowe run interpreters for their DSL in
+  a Plutus script. Execution limits currently restrict the complexity of the DSL
+  expressions that can be evaluated in a single transaction.
+
+<a id="novel-user-cases" href="#novel-user-cases">**Novel use cases**</a>
+
+Although the version of Leios proposed in this document does not support the
+particular use cases listed below, a minor variant or future version of Leios
+could.
+
+- _Priority pipelines:_ Different Leios pipelines might have different stage
+  lengths, throughput, fees, and/or Plutus execution limits, enabling
+  applications to select their level of service.
+- _Externally batched endorser blocks:_ Third parties could construct endorser
+  blocks and provide them directly to the block producers, allowing a dapp or an
+  exchange detailed control over sequencing of interdependent transactions
+  within the endorser block.
+- _Nuanced roles for SPOs:_ Leios opens the possibility of separating the
```

Comment:
Technically, you could already achieve this by adding a tag in the lottery check (the sortition paper already mentions this). You could hence say that the primitives can support in the future such functionality but this has not been considered at the moment.